pkg/users/account-create-dialog.js

```diff
@@ -169,31 +217,59 @@
     return remove_diacritics(result);
 }
 
-export function account_create_dialog(accounts) {
+export function account_create_dialog(accounts, shells) {
     let dlg = null;
     const state = {
+        dialogLoading: true,
         real_name: "",
         user_name: "",
         password: "",
         password_confirm: "",
         locked: false,
         confirm_weak: false,
         change_passw_force: false,
+        base_home_dir: "",
+        home_dir: "",
+        home_dir_dirty: false,
+        shell: "",
+        isShellSelectExpanded: false,
     };
     let errors = { };
 
     let old_password = null;
     let user_name_dirty = false;
 
+    function get_defaults() {
+        return cockpit.spawn(["/usr/sbin/useradd", "-D"], { superuser: "require" })
+                .catch(() => "")
+                .then(defaults => {
+                    defaults.split("\n").forEach(item => {
+                        if (item.indexOf("SHELL=") === 0) {
+                            change("shell", item.split("=")[1] || "/bin/bash");
+                        } else if (item.indexOf("HOME=") === 0) {
+                            change("base_home_dir", item.split("=")[1] || "");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230131-135835-5b43f8e5-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)